frame/evm/precompile/contracts/transfer/src/lib.rs

```diff
@@ -0,0 +1,71 @@
+// This file is part of Darwinia.
+//
+// Copyright (C) 2018-2021 Darwinia Network
+// SPDX-License-Identifier: GPL-3.0
+//
+// Darwinia is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Darwinia is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Darwinia. If not, see <https://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub mod kton;
+pub mod ring;
+pub mod util;
+
+// --- substrate ---
+use sp_std::{marker::PhantomData, vec::Vec};
+// --- darwinia ---
+use darwinia_support::evm::SELECTOR;
+use dp_evm::Precompile;
+use kton::Kton;
+use ring::RingBack;
+// --- crate ---
+use evm::{Context, ExitError, ExitSucceed};
+
+pub type AccountId<T> = <T as frame_system::Config>::AccountId;
+
+/// Transfer Precompile Contract, used to support the exchange of KTON and RING tranfer.
+pub enum Transfer<T> {
+	/// Transfer RING back from DVM to Darwinia
+	RingBack,
```

Comment:
Duplicate name with ```ring::RingBack```

Reply:
Hmm, change to `RingTransfer` would be better. Let me fix it.